### PR TITLE
Fixed Forms for Job Posts and Create Profile

### LINKED ISSUE #9: Add Forms for Jobs, Gigs, and Profile Creation

### PR DESCRIPTION
**Pull request by angeljgomezc**
### What is this Pull Request?

It contains new inputs, components, modals, and a reorganization of files. There's also a bit of redesign. 

### What does it fix?

This is all to make the `/create-profile` and `post-job` forms work and make them readily available for #76 issue. 

### Does it close any issue?

Yes, it closes #9 (or at least, it should for now)

### Does it need a code review?

Yes, I invoke @with-heart @Dhaiwat10 and @carlomigueldy to check it out

### CODE REVIEW

File: client/components/forms/post-job/3_JobSummary.tsx [hunk 1]
```diff
@@ -90,53 +93,49 @@ export default function GigSummary({
...
-        </HStack>
+      <Stack spacing={2} p={2} maxW="100%">
+        <Heading fontSize="sm">Keywords:</Heading>
+
+        <KeywordsSection
+          keywordsData={jobKeywords}
+          templateColumns="repeat(3, 3fr)"
+        />
+
+        <br />
```
Comment (frankTurtle) [inline client/components/forms/post-job/3_JobSummary.tsx:112]: Not sure why, but whenever I see these `<br />` I always chuckle and think about the old school `<blink></blink>` tags. No matter how far we've come in webdev, these bad boys will never die haha.

File: client/components/forms/post-job/3_JobSummary.tsx [hunk 2]
```diff
@@ -66,16 +66,19 @@ export default function GigSummary({
         <Text>Check that everything is correct (click edit if not)</Text>
       </Container>
 
-      <Container margin="auto" textAlign="center" maxW="100%">
-        <Heading fontSize="lg">Job Basics</Heading>
-        <ButtonGray onClick={goToBasics}>Edit</ButtonGray>
+      <Container m="auto" textAlign="center" maxW="100%">
+        <Heading fontSize="lg">
+          Job Basics <ButtonGray onClick={goToBasics}>Edit</ButtonGray>
```
Comment (frankTurtle) [inline client/components/forms/post-job/3_JobSummary.tsx:71]: Wasn't there a str localization implementation somewhere?

File: client/components/modals/AddLinks.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                {' '}
+                <FontAwesomeIcon icon={faDribbble} />
+              </InputLeftAddon>
+              <Input
+                placeholder="username"
+                name="dribbble"
+                value={dribbble}
+                onChange={(e) => onChange(e)}
+              ></Input>
+            </InputGroup>
```
Comment (frankTurtle) [inline client/components/modals/AddLinks.tsx:148]: When I see so much duplicate code with such small variation, I feel like it should be a loop/map

File: client/components/modals/SelectKeywords.tsx [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    const isInArray = selectedKeywords.find(
+      (element: any) => element.keyword === e
+    );
+    if (isInArray || selectedKeywords.length >= 10) {
+      return;
+    }
+    //Merges the selected keywords with the state
+    setSelectedKeywords((prevKeywords: any) => {
+      const updatedKeywords: any = [...prevKeywords];
+      updatedKeywords.unshift({ keyword: e, id: Math.random().toString() });
```
Comment (frankTurtle) [inline client/components/modals/SelectKeywords.tsx:65]: Curious what the `id` part of this object is for? 🤔

File: client/components/modals/SelectKeywords.tsx [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+                    .toLowerCase()
+                    .includes(searchKeywords.toLowerCase())
+                ) {
+                  return keyword;
+                }
+                {
+                  return false;
+                }
+              })
+              .map((keyword, index) => (
```
Comment (frankTurtle) [inline client/components/modals/SelectKeywords.tsx:151]: Aah, I think this answers my previous question about the `id`.

**Review by frankTurtle — COMMENTED** [review]
🤙🤙

File: client/components/modals/SelectKeywords.tsx [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+  }, [selectedKeywords, keywordsData]);
+
+  const selectKeyword = (e: string) => {
+    //Limits the number of selected keywords to 10
+    const isInArray = selectedKeywords.find(
+      (element: any) => element.keyword === e
+    );
+    if (isInArray || selectedKeywords.length >= 10) {
+      return;
+    }
```
Comment (frankTurtle) [inline client/components/modals/SelectKeywords.tsx:61]: Shouldn't this just be?
`if (isInArray || selectedKeywords.length > 10) { return; }`

Having it `=== 10` is desired, no?

File: package.json [hunk 7]
```diff
@@ -29,5 +29,13 @@
     "*.{ts,tsx,md,json}": [
       "prettier --write"
     ]
+  },
+  "dependencies": {
+    "@chakra-ui/icons": "^1.1.1",
```
Comment (JoviDeCroock) [inline package.json:34]: This belongs in client/package.json

File: client/pages/post-job/index.tsx [hunk 8]
```diff
@@ -34,13 +21,8 @@ export default function CreateProject() {
...
 
-  const [timeframeActive, setTimeframeActive] = useState(false);
-
-  const [locationActive, setLocationActive] = useState(false);
-
-  const addLocation = () => {
-    setLocationActive(true);
-  };
+  //Job Keywords
+  const [jobKeywords, setJobKeywords] = useState<any>();
```
Comment (JoviDeCroock) [inline client/pages/post-job/index.tsx:25]: Shouldn't this be a string[] instead of any

File: client/pages/create-profile/index.tsx [hunk 9]
```diff
@@ -42,10 +44,65 @@ export default function CreateProfile() {
...
+  };
+
+  const closeLinksModal = () => {
+    setChangeProfileLinks(false);
+  };
+
+  //Sets links data and sends to database
+  const linksDataHandler = async (linkData: any) => {
+    await setProfileLinks(linkData);
+    console.log(profileLinks);
```
Comment (JoviDeCroock) [inline client/pages/create-profile/index.tsx:104]: Log

File: client/pages/create-profile/index.tsx [hunk 10]
```diff
@@ -42,10 +44,65 @@ export default function CreateProfile() {
...
+  };
+
+  const closeKeywordModal = () => {
+    setChangeProfileKeywords(false);
+  };
+
+  //Sets profile keywords data
+  const keywordsDataHandler = async (selectedKeywords: any) => {
+    await setProfileKeywords(selectedKeywords);
+    console.log(profileKeywords);
```
Comment (JoviDeCroock) [inline client/pages/create-profile/index.tsx:87]: Log

File: client/components/modals/SelectKeywords.tsx [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+
+  const onSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    keywordsDataHandler(selectedKeywords);
+    closeKeywordModal();
+  };
+
+  return (
+    <>
+      <Box
```
Comment (JoviDeCroock) [inline client/components/modals/SelectKeywords.tsx:89]: Why are we using an empty box, I think an on-click-outside listener is going to be a lot easier

File: client/components/modals/SelectKeywords.tsx [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+
+export default function KeywordSelect({
+  keywordsDataHandler,
+  closeKeywordModal,
+  keywordsData,
+}: KeywordProps): any {
+  let keywords = keywordsSamples;
+
+  const [searchKeywords, setSearchKeywords] = useState('');
+  const [selectedKeywords, setSelectedKeywords] = useState<any>([]);
```
Comment (JoviDeCroock) [inline client/components/modals/SelectKeywords.tsx:43]: Let's avoid any

File: client/components/create-profile/LinksSection.tsx [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+                >
+                  <TagLabel m="auto">
+                    <FontAwesomeIcon icon={faDribbble} />
+                  </TagLabel>
+                </Tag>
+              </Link>
+            </NextLink>
+          ) : null}
+
+          {profileLinks && behance ? (
```
Comment (JoviDeCroock) [inline client/components/create-profile/LinksSection.tsx:102]: We already check profileLinks in the wrapping entity

**Review by JoviDeCroock — CHANGES_REQUESTED** [review]
Some residual logs and type-safety

File: client/components/modals/SelectKeywords.tsx (hunk 6 above)
Comment (frankTurtle) [reply client/components/modals/SelectKeywords.tsx:61]: Assuming my thoughts are incorrect then?

File: client/components/Landing.tsx [hunk 14]
```diff
@@ -1,13 +1,12 @@
...
 
 import NextLink from 'next/link';
 
 import {
   ButtonRed,
-  ButtonBlack,
   ButtonGreen,
   ButtonBlue,
-} from '@/styles/ui-components/Chakra-Button';
+} from '../styles/ui-components/Chakra-Button';
```
Comment (frankTurtle) [inline client/components/Landing.tsx:9]: Path alias not working?

File: .gitignore [hunk 15]
```diff
@@ -1,4 +1,5 @@
 # See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+/TESTS
```
Comment (frankTurtle) [inline .gitignore:2]: Which `/TESTS` is this ignoring? I'm assuming it's not unit tests or regular tests, right?

File: client/components/forms/post-job/1_JobBasics.tsx [hunk 16]
```diff
@@ -1,28 +1,33 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { GetStaticProps } from 'next';
 
 // UI & CSS
 import {
   ButtonGreen,
   ButtonOrange,
-} from '@/styles/ui-components/Chakra-Button';
+} from '../../../styles/ui-components/Chakra-Button';
```
Comment (frankTurtle) [inline client/components/forms/post-job/1_JobBasics.tsx:8]: Aliases again? .... 🤔

**Review by frankTurtle — CHANGES_REQUESTED** [review]
Something is FUBAR if you've to revert all the path aliases

Is it throwing an error, or not importing them properly?

It's inconsistent with using vs not using them, too
* ex -> `client/components/developers/DeveloperItem.tsx` uses it just fine

File: client/components/create-profile/LinksSection.tsx [hunk 17]
```diff
@@ -0,0 +1,151 @@
...
+
+export default function LinksSection({ profileLinks }: profileLinksProps) {
+  const { linkedin, twitter, behance, dribbble, producthunt, github } =
+    profileLinks;
+
+  return (
+    <>
+      {profileLinks && (
+        <HStack mt="15px" gap="0.5rem" spacing={1}>
+          {github ? (
```
Comment (frankTurtle) [inline client/components/create-profile/LinksSection.tsx:33]: A lot of duplicate code - Seems like another candidate for a map/loop

File: client/components/forms/post-job/2_JobDetails.tsx [hunk 18]
```diff
@@ -1,72 +1,107 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { GetStaticProps } from 'next';
 
+//Keywords Components
+import KeywordSelect from '../../modals/SelectKeywords';
```
Comment (frankTurtle) [inline client/components/forms/post-job/2_JobDetails.tsx:5]: Missing Alias

File: client/components/forms/post-job/2_JobDetails.tsx [hunk 19]
```diff
@@ -1,72 +1,107 @@
...
+import KeywordsSection from '../KeywordsSection';
+
 import {
   ButtonGreen,
   ButtonOrange,
   ButtonBlack,
-} from '@/styles/ui-components/Chakra-Button';
+} from '../../../styles/ui-components/Chakra-Button';
```
Comment (frankTurtle) [inline client/components/forms/post-job/2_JobDetails.tsx:12]: Missing path alias

File: client/components/jobs/JobItem.tsx [hunk 20]
```diff
@@ -27,7 +27,7 @@ export default function JobItem() {
     >
       <Box ml="0.5%" maxW="100%">
         <Link
-          href="/jobs/project-1"
+          href="/jobs/username/marketing-manager"
```
Comment (frankTurtle) [inline client/components/jobs/JobItem.tsx:30]: Not sure if we use these paths anywhere else, but if we do, we should abstract them out and import into each spot so we only have to change it once in a central location

A constants file maybe?

File: client/components/modals/AddLinks.tsx [hunk 21]
```diff
@@ -0,0 +1,220 @@
...
+  VStack,
+  InputGroup,
+  InputLeftAddon,
+  ButtonGroup,
+} from '@chakra-ui/react';
+
+import {
+  ButtonBlue,
+  ButtonOrange,
+} from '../../styles/ui-components/Chakra-Button';
```
Comment (frankTurtle) [inline client/components/modals/AddLinks.tsx:16]: Alias

File: client/components/modals/AddLinks.tsx [hunk 22]
```diff
@@ -0,0 +1,220 @@
...
+//     <FontAwesomeIcon icon={faDribbble} />
+//   </InputLeftAddon>
+//   <Input
+//     placeholder="username"
+//     name="dribbble"
+//     value={dribbble}
+//     onChange={(e) => onChange(e)}
+//   ></Input>
+// </InputGroup>
+//
```
Comment (frankTurtle) [inline client/components/modals/AddLinks.tsx:220]: Remove

File: client/components/modals/SelectKeywords.tsx [hunk 23]
```diff
@@ -0,0 +1,213 @@
...
+  TagLabel,
+  TagCloseButton,
+  TagRightIcon,
+  ButtonGroup,
+} from '@chakra-ui/react';
+
+import {
+  ButtonBlue,
+  ButtonOrange,
+} from '../../styles/ui-components/Chakra-Button';
```
Comment (frankTurtle) [inline client/components/modals/SelectKeywords.tsx:25]: Alias

File: client/components/modals/SelectKeywords.tsx [hunk 24]
```diff
@@ -0,0 +1,213 @@
...
+      }
+    }
+  }, [selectedKeywords, keywordsData]);
+
+  const selectKeyword = (e: string) => {
+    //Limits the number of selected keywords to 10
+    const isInArray = selectedKeywords.find(
+      (element: Keywords) => element.keyword === e
+    );
+    if (isInArray || selectedKeywords.length > 10) {
```
Comment (frankTurtle) [inline client/components/modals/SelectKeywords.tsx:65]: 🎉

File: client/components/navigation/Navbar.tsx [hunk 25]
```diff
@@ -19,7 +15,7 @@ import { useEthers } from '@usedapp/core';
 
 import { HamburgerIcon, CloseIcon } from '@chakra-ui/icons';
 
-import { ButtonBlack } from '@/styles/ui-components/Chakra-Button';
+import { ButtonBlack } from '../../styles/ui-components/Chakra-Button';
```
Comment (frankTurtle) [inline client/components/navigation/Navbar.tsx:18]: Alias

File: client/components/jobs/JobItem.tsx (hunk 20 above)
Comment (with-heart) [reply client/components/jobs/JobItem.tsx:30]: I agree with this in theory but I think we should wait till it becomes an actual problem for us, like we should wait till we've had to fix a broken link or three and then we can figure out an abstraction for that. Makes sure we understand the pain(s) we're solving for before we abstract.

File: .gitignore (hunk 15 above)
Comment (with-heart) [reply .gitignore:2]: Would love to know this as well! If it's some type of local folder you're using to test things, there's probably a library we could use to automate whatever it is you're doing!

**Pull request merged**